setup.sh

```diff
@@ -0,0 +1,617 @@
+    echo
+    writeBold "$RED" "[✖] Ooops! Something went wrong."
+    write            "    Kuzzle does not seem to respond as expected to requests to"
+    write            "    http://localhost:7511"
+    echo
+    writeBold "$YELLOW" "Feel free to get in touch with the support team by sending"
+    writeBold "$YELLOW" "a mail to $SUPPORT_MAIL or by joining the chat room on"
+    writeBold "$YELLOW" "Gitter at $GITTER_URL - We'll be glad to help you."
+    echo
+    write     "Sorry for the inconvenience."
+    echo
+    exit 8
+  else
+    writeBold "$GREEN" "[✔] Kuzzle is running!"
+  fi
+}
+
+shortHelp() {
+  echo
+  writeBold "* You can open this short help by calling ./setup.sh --help"
+  writeBold "* You can start Kuzzle by typing:"
+  write "  docker-compose -f $composerYMLPath up -d"
+  writeBold "* You can see the logs of the Kuzzle stack by typing:"
+  write "  docker-compose -f $composerYMLPath logs -f"
+  writeBold "* You can check if everything is working by typing:"
+  write "  curl -XGET http://localhost:7512/"
+  writeBold "* You can stop Kuzzle by typing:"
+  write "  docker-compose -f $composerYMLPath stop"
+  writeBold "* You can restart Kuzzle by typing:"
+  write "  docker-compose -f $composerYMLPath restart"
+  writeBold "* You can read the docs at http://docs.kuzzle.io/"
+  echo
+}
+
+# Main execution routine
+# ===========================
+
+if [ "$1" == "--help" ]; then
+  echo
+  writeBold "# Kuzzle short help"
+  writeBold "  ================="
+  shortHelp
+  exit 0
+fi
+
+clear
+echo
+writeBold "# Kuzzle Setup"
+writeBold "  ============"
+echo
+writeBold "This script will help you launching Kuzzle and installing"
+writeBold "all the necessary dependencies."
+echo
+write     "* You can refer to http://docs.kuzzle.io/ if you need better"
+write     "  understanding of the installation process."
+write     "* Feel free to join us on Gitter at $GITTER_URL"
+write     "  if you need help."
+echo
+promptBold "$BLUE" "Press Enter to start."
+read start
+
+echo
+writeBold "[ℹ] Checking system pre-requisites..."
+echo
+
+CHECK_ARCH=$(uname -a | grep x86_64)
+if [ -z "${CHECK_ARCH}" ]; then
+  echo
+  writeBold "$RED" "[✖] Kuzzle runs on x86_64 architectures, which does not seem"
+  writeBold "$RED" "    to be the architecture of your system."
+  write            "    Sorry, you cannot launch Kuzzle on this machine."
+  echo
+  exit 4
+fi
+
+write "$GREEN" "[✔] Architecture is x86_64."
+
+CHECK_MEM=$(awk '/MemTotal/{print $2}' /proc/meminfo)
+MEM_REQ=4000000
+if (( "$CHECK_MEM" < "$MEM_REQ" )); then
+  echo
+  writeBold "$RED" "[✖] Kuzzle needs at least 4Gb of memory, which does not seem"
+  writeBold "$RED" "    to be the available amount on your system (${CHECK_MEM})."
+  write            "    Sorry, you cannot launch Kuzzle on this machine."
+  echo
+  exit 5
+fi
+
+write "$GREEN" "[✔] Available memory is at least 4Gb."
+
+CHECK_CORES=$(awk '/^processor/{print $3}' /proc/cpuinfo | tail -1)
+MIN_CORES=4
```

Comment:
Ok, ES recommends to use a multi-core processor, but enforces no rule about the number of cores.

> You should choose a modern processor with multiple cores. 

https://www.elastic.co/guide/en/elasticsearch/guide/current/hardware.html

But it indeed works with just one processor, so I'm killing the whole block.